xrpl/models/transactions/deposit_preauth.py

```diff
@@ -39,14 +41,90 @@ class DepositPreauth(Transaction):
         init=False,
     )
 
+    authorize_credentials: Optional[List[Credential]] = None
+    """The credential(s) that received the preauthorization. (Any account with these
+    credentials can send preauthorized payments)."""
+
+    unauthorize_credentials: Optional[List[Credential]] = None
+    """The credential(s) whose preauthorization should be revoked."""
+
     def _get_errors(self: Self) -> Dict[str, str]:
         errors = super()._get_errors()
-        if self.authorize and self.unauthorize:
-            errors[
-                "DepositPreauth"
-            ] = "One of authorize and unauthorize must be set, not both."
 
-        if not self.authorize and not self.unauthorize:
-            errors["DepositPreauth"] = "One of authorize and unauthorize must be set."
+        if (
+            self.authorize is None
+            and self.unauthorize is None
+            and self.authorize_credentials is None
+            and self.unauthorize_credentials is None
+        ):
+            errors["DepositPreauth"] = (
+                "Exactly one input parameter amongst authorize, unauthorize, "
+                + "authorize_credentials or unauthorize_credentials must be set. It is "
+                + "invalid if none of the params are specified."
+            )
+
+        if (
+            sum(
```

Comment:
```suggestion
            len(
```

Reply:
f2651ca2

Reply:
Note: I cannot accept this code-suggestion because it doesn't work in the given context.
`len([True, False, False, False]) == 4`, it does not capture the fact that only one of the elements is True.

However, `sum([True, False, False, False]) == 1`

Reply:
It's pretty un-Pythonic (and unintuitive for Python devs) to sum booleans. You can add an `if param is not False` if you want.

Reply:
`if param is not False` does not work. It will need to be `if refined_param is True for refined_param in [if param is not None for param in inputs_params]` -- I can't get around this nested loop comprehension.

I came with a `filter` construct 07191079 -- this is easier to understand